CRM/Core/Form.php

```diff
@@ -179,6 +179,29 @@ class CRM_Core_Form extends HTML_QuickForm_Page {
    */
   public $urlPath = array();
 
+  /**
+   * Context of the form being loaded.
+   *
+   * 'event' or null
+   *
+   * @var string
+   */
+  protected $context;
+
+  /**
+   * @return string
+   */
+  public function getContext() {
+    return $this->context;
+  }
+
+  /**
+   * Set context variable.
+   */
+  public function setContext() {
+    $this->context = CRM_Utils_Request::retrieve('context', 'String', $this);
```

Comment:
Change to `Alphanumeric`?

Reply:
@colemanw I would love to but the Alphanumeric rule needs to be merged first - I did include in this PR in the hope we could start using it - https://github.com/civicrm/civicrm-core/pull/12185

Reply:
@eileenmcnaughton that PR has now been merged, so this can be updated.

Reply:
@colemanw fixed